File: backend-dotnet/Models/Job.cs
namespace backend_dotnet.Models;

public class Job
{
    public Guid JobId { get; set; } = Guid.NewGuid();
    public Guid CompanyId { get; set; }
    public string JobTitle { get; set; } = string.Empty;
    public string TargetDomain { get; set; } = string.Empty;
    public string JobDescriptionSummary { get; set; } = string.Empty;
    public string[] InternshipType { get; set; } = Array.Empty<string>();
    public string LocationCity { get; set; } = string.Empty;
    public decimal MinimumGPA { get; set; }
    public int[] AllowedYearsOfStudy { get; set; } = Array.Empty<int>();
    public string[] MandatorySkills { get; set; } = Array.Empty<string>();
    public string[] NiceToHaveSkills { get; set; } = Array.Empty<string>();
    public string[] PreferredDegreePrograms { get; set; } = Array.Empty<string>();
    public bool StipendOffered { get; set; }
    public string? StipendAmountOrDetails { get; set; }
    public int DurationMonths { get; set; }
    public DateTime ApplicationDeadline { get; set; }

    // Navigation properties
    public CompanyProfile Company { get; set; } = null!;
    public ICollection<Application> Applications { get; set; } = new List<Application>();
}

File: frontend-react/src/types/job.ts
export interface TargetDomain {
  id: number;
  name: string;
  description?: string;
  titleCount: number;
}

export interface JobTitle {
  id: number;
  title: string;
  targetDomainId: number;
  targetDomainName: string;
}

export type InternshipTypeOption = 'OnSite' | 'Hybrid' | 'Remote';

export interface CreateJobPayload {
  companyId?: string;
  recruiterEmail?: string;
  targetDomain: string;
  targetDomainId?: number;
  jobTitle: string;
  jobTitleId?: number;
  jobDescriptionSummary: string;
  internshipType: string;
  locationCity: string;
  minimumGPA: number;
  allowedYearsOfStudy: number[];
  mandatorySkills: string[];
  niceToHaveSkills: string[];
  preferredDegreePrograms: string[];
  stipendOffered: boolean;
  stipendAmountOrDetails?: string;
  durationMonths: number;
  applicationDeadline: string;
}

export interface JobResponse {
  jobId: string;
  companyId: string;
  companyName: string;
  jobTitle: string;
  targetDomain: string;
  jobDescriptionSummary: string;
  internshipType: string[];
  locationCity: string;
  minimumGPA: number;
  allowedYearsOfStudy: number[];
  mandatorySkills: string[];
  niceToHaveSkills: string[];
  preferredDegreePrograms: string[];
  stipendOffered: boolean;
  stipendAmountOrDetails?: string;
  durationMonths: number;
  applicationDeadline: string;
  matchesVerified: number;
  status: string;
}
